### PR TITLE
RevEng: Reuse ICSharpHelper

### LINKED ISSUE #10134: RevEng: Multi-line DEFAULT SQL results in invalid C#
Describe what is not working as expected.
When creating the model for table columns that have a default value set via a function, the scaffolding is creating incorrect syntax in the ModelCreating.

In sql server, I have a column with a "Default Vale or Binding" set as `dbo.Set_To_Zero`.

The resulting code created in `protected override void OnModelCreating(ModelBuilder modelBuilder)` was:

``` C#
.HasDefaultValueSql("
/****** Object:  Default dbo.Set_To_Zero    Script Date: 8/20/01 1:54:04 PM ******/

/****** Object:  Default dbo.Set_To_Zero    Script Date: 8/6/01 11:18:06 AM ******/

CREATE DEFAULT Set_To_Zero
\tAS 0




");
```

### Steps to reproduce
Have a sql server column with a Default Value or Binding set to a db function. 
Use DB Scaffold-DbContext to have it create the models.
View result in the resulting Context.cs file.

### Further technical details
EF Core version: (found in project.csproj or packages.config)
    ```
<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="1.1.2" PrivateAssets="All" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="1.1.2" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer.Design" Version="1.1.2" PrivateAssets="All" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="1.1.1" PrivateAssets="All" />
```
Database Provider:  Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 x64
IDE: e.g. Visual Studio 2017

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #10134, part of #8853

### CODE REVIEW

File: src/EFCore.Design/Design/ICSharpHelper.cs [hunk 1]
```diff
@@ -12,6 +12,13 @@ namespace Microsoft.EntityFrameworkCore.Design
     /// </summary>
     public interface ICSharpHelper
     {
+        /// <summary>
+        ///     Generates a method call code fragment.
+        /// </summary>
+        /// <param name="fragment"> The method call. </param>
+        /// <returns> The fragment. </returns>
+        string Fragment([NotNull] MethodCallCodeFragment fragment);
```
Comment (smitpatel) [inline src/EFCore.Design/Design/ICSharpHelper.cs]: breaking changes?

Comment (smitpatel) [reply src/EFCore.Design/Design/ICSharpHelper.cs]: Should name this method something more descriptive

Comment (bricelam) [reply src/EFCore.Design/Design/ICSharpHelper.cs]: Will review the breaking changes and see if there's a non-sql breaking way to do it. Overload resolution might get ugly. Break is probably very low impact, but I'll do my due diligence.

Comment (bricelam) [reply src/EFCore.Design/Design/ICSharpHelper.cs]: Name is intentionally vague. I imagine more overloads as we need to generate different kinds of fragments. This is a mostly internal DSL class so the names are a bit unorthodox

Comment (bricelam) [reply src/EFCore.Design/Design/ICSharpHelper.cs]: I like to think of it as analogous to the Html helper in Razor.

**Comment by bricelam** [comment]
Breaking changes reverted (except for the added interface member)

File: test/EFCore.Design.Tests/Design/Internal/CSharpHelperTest.cs [hunk 2]
```diff
@@ -261,6 +261,59 @@ public void Namespace_works(string[] input, string excepted)
...
+        }
+
+        [Fact]
+        public void Fragment_MethodCallCodeFragment_works_with_arrays()
+        {
+            var method = new MethodCallCodeFragment("Test", new byte[] { 1, 2 }, new[] { 3, 4 }, new[] { "foo", "bar" });
+
+            var result = new CSharpHelper().Fragment(method);
+
+            Assert.Equal(".Test(new byte[] { 1, 2 }, new[] { 3, 4 }, new[] { \"foo\", \"bar\" })", result);
```
Comment (bricelam) [inline test/EFCore.Design.Tests/Design/Internal/CSharpHelperTest.cs]: cc @roji No more `int[]` or `string[]`. Just `byte[]`. 😉

File: src/EFCore.Design/breakingchanges.netcore.json [hunk 3]
```diff
@@ -1,4 +1,9 @@
 [
+  {
+    "TypeId": "public interface Microsoft.EntityFrameworkCore.Design.ICSharpHelper",
+    "MemberId": "System.String Fragment(Microsoft.EntityFrameworkCore.Design.MethodCallCodeFragment fragment)",
+    "Kind": "Addition"
```
Comment (bricelam) [inline src/EFCore.Design/breakingchanges.netcore.json]: cc @ajcvickers

File: test/EFCore.Design.Tests/Design/Internal/CSharpHelperTest.cs (hunk 2 above)
Comment (roji) [reply test/EFCore.Design.Tests/Design/Internal/CSharpHelperTest.cs]: Nice :)

**Pull request merged**